emission/analysis/modelling/tour_model/similarity.py

```diff
@@ -52,14 +61,16 @@ def __init__(self, data, radius):
                 end_lat = end_place.data.location["coordinates"][1]
                 logging.debug("endpoints are = (%s, %s) and (%s, %s)" %
                               (start_lon, start_lat, end_lon, end_lat))
-                if self.distance(start_lat, start_lon, end_lat, end_lon):
-                    self.data.remove(t)
+                if similarity(data, radius).distance(start_lat, start_lon, end_lat, end_lon):
+                    data.remove(t)
```

Comment:
wait, you are creating a similarity object to compute the distance between two points?!!